hpms/src/main/java/gov/cms/ab2d/hpms/service/HPMSAuthServiceImpl.java

```diff
@@ -105,17 +107,26 @@ private void refreshToken(long currentTimestamp) {
             tokenExpires = currentTimestamp + authResponse.getExpires() * 900L;
             authToken = authResponse.getAccessToken();
         } catch (WebClientResponseException exception) {
-            eventLogger.log(EventClient.LogType.SQL,
-                    new ErrorEvent(HPMS_ORGANIZATION, "", HPMS_AUTH_ERROR, prepareErrorMessage(exception, curTime)));
+            logEvents(prepareErrorMessage(exception, curTime));
             throw exception;
         } catch (IllegalStateException | NullPointerException exception) {
-            String message = "HPMS auth call failed with no response waited for " + (curTime / 1000) + " seconds.";
-            eventLogger.log(EventClient.LogType.SQL,
-                    new ErrorEvent(HPMS_ORGANIZATION, "", HPMS_AUTH_ERROR, message));
-            throw new RemoteTimeoutException(message);
+            String error = "HPMS auth call failed with no response waited for " + (curTime / 1000) + " seconds.";
+            logEvents(error);
+            throw new RemoteTimeoutException(error);
         }
     }
 
+    private void logEvents(String error) {
+        eventLogger.sendLogs(MetricsEvent.builder()
+                .service("HPMS")
+                .timeOfEvent(OffsetDateTime.now())
+                .eventDescription(String.format(error))
+                .stateType(MetricsEvent.State.CONTINUE)
+                .build());
+        eventLogger.log(EventClient.LogType.SQL,
```

Comment:
I can remove this one. I added this so we can track hpms errors but we don't really need to track it twice.

Reply:
Yes, sendLogs method already logs to sql.